src/cli/mod.rs

```diff
@@ -48,12 +52,82 @@ pub struct Options {
 
 impl Options {
     pub fn execute(self) -> anyhow::Result<ExitCode> {
-        self.logger.init();
+        let env_filter = EnvFilter::builder()
+            .with_default_directive(self.logger.verbosity_level().into())
+            .parse_lossy(self.logger.log_filter.as_ref().unwrap_or(&"".to_owned()));
+        #[cfg(unix)]
+        let log_level = env_filter
+            .max_level_hint()
+            .and_then(LevelFilter::into_level)
+            .map(Into::into);
+        let fmt_layer = tracing_subscriber::fmt::layer()
+            .with_writer(move || -> Box<dyn io::Write> {
+                match self.logger.log_target {
+                    LogTarget::Stdout => Box::new(io::stdout()),
+                    LogTarget::Stderr => Box::new(io::stderr()),
+                }
+            })
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .with_filter(env_filter);
+
+        #[cfg(feature = "bench")]
+        let (flame_layer, _guard, profile) = if let Some(ref profile) = self.logger.profile {
```

Comment:
Same suggestion for `match` as above.

Reply:
same as above, I don't think we need to add another indentation level here. what's the benefit?